src/OpenTelemetry.AutoInstrumentation.StartupHook/RulesEngine/ApplicationInExcludeListRule.cs

```diff
@@ -55,12 +70,18 @@ private static string GetApplicationName()
         }
     }
 
-    private static bool IsApplicationInExcludeList(string applicationName)
+    private static bool IsApplicationInExcludeList(string appDomainName, string processModuleName)
+    {
+        return GetExcludedApplicationNames().Contains(processModuleName, StringComparer.InvariantCultureIgnoreCase) ||
+            GetExcludedAppDomainNames().Contains(appDomainName, StringComparer.InvariantCultureIgnoreCase);
```

Comment:
Is this documented?

Reply:
@pellared any ideas where to put it? - 99.9% does not concern end users

Reply:
At least comment in code 😉

Reply:
https://github.com/open-telemetry/opentelemetry-dotnet-instrumentation/pull/2765#discussion_r1269972485

Separated checks, I guess the `Logger.Information($"Rule Engine: AppDomain name is dotnet. Skipping initialization.")` acts as a replacement for a comment now 🤔